packages/typescript/src/browser/typescript-client-contribution.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { BaseLanguageClientContribution, Workspace, Languages, LanguageClientFactory } from '@theia/languages/lib/browser';
+import { TYPESCRIPT_LANGUAGE_ID, TYPESCRIPT_LANGUAGE_NAME } from '../common';
+import { registerTypeScript } from './typescript-language-config';
+
+@injectable()
+export class TypescriptClientContribution extends BaseLanguageClientContribution {
+
+    readonly id = TYPESCRIPT_LANGUAGE_ID;
+    readonly name = TYPESCRIPT_LANGUAGE_NAME;
+
+    constructor(
+        @inject(Workspace) protected readonly workspace: Workspace,
+        @inject(Languages) protected readonly languages: Languages,
+        @inject(LanguageClientFactory) protected readonly languageClientFactory: LanguageClientFactory
+    ) {
+        super(workspace, languages, languageClientFactory);
+        registerTypeScript();
+    }
+
+    protected get globPatterns() {
+        return [
+            '**/*.ts',
```

Comment:
What about javascript? I think it was covered by the typescript package as well.

Reply:
Should we add it to the typescript extension? It felt a bit odd. OTOH there are lots of similarities :).

Reply:
in VS code it seems to be 2 different extensions, I wonder how they handle the case when tsconfig allow compiling js files